### ISSUE TITLE
`XhtmlType` should be a `PrimitiveType`

### ISSUE THREAD

**Issue by bashir2** [hapifhir/org.hl7.fhir.core#1800]
This is based on the discussion [here](https://github.com/hapifhir/org.hl7.fhir.core/issues/1678#issuecomment-2455445033):

In R4 `XhtmlType` is not a `PrimitiveType` ([link](https://github.com/hapifhir/org.hl7.fhir.core/blob/c06051838f9bcd5fc0b552df960f5bf2fcf2dbb4/org.hl7.fhir.r4/src/main/java/org/hl7/fhir/r4/model/XhtmlType.java#L40)), but in R5 it is ([link](https://github.com/hapifhir/org.hl7.fhir.core/blob/c06051838f9bcd5fc0b552df960f5bf2fcf2dbb4/org.hl7.fhir.r5/src/main/java/org/hl7/fhir/r5/model/XhtmlType.java#L43)). This seems to be a bug in R4 because the corresponding type for `xhtml` elements in R4 StructureDefinitions is [`RuntimePrimitiveDatatypeXhtmlHl7OrgDefinition`](https://github.com/hapifhir/hapi-fhir/blob/a52578e798c0f32eafb9804719bb705394029260/hapi-fhir-base/src/main/java/ca/uhn/fhir/context/RuntimePrimitiveDatatypeXhtmlHl7OrgDefinition.java#L27) which is a `BaseRuntimeElementDefinition<IBaseXhtml>` and `IBaseXhtml` is an `IPrimitiveType`.

**Comment by grahamegrieve** [comment]
@jamesagnew does HAPI have any opinion on the answer to this issue?

**Comment by bashir2** [comment]
This is still an issue for us as documented [here](https://github.com/google/fhir-data-pipes/issues/1014); so it would be nice if it can be fixed (it seems I cannot reopen the issue).

**Comment by dotasek** [comment]
According to the spec, the core library has this correct:

https://www.hl7.org/fhir/R4/datatypes.html xhtml - Element
https://www.hl7.org/fhir/R5/datatypes.html xhtml - DataType

In HAPI, I think this would mean creating a `RuntimeElementXhtmlHl7OrgDefinition` as well as code to handle it in ModelScanner and other places. @jamesagnew, does this mean opening a HAPI ticket?

**Comment by jamesagnew** [comment]
Where does XhtmlType actually get used? As far as I can see, we use the exact same XhtmlNode class in both the R4 and R5 Narrative typers.

**Comment by grahamegrieve** [comment]
We use in the core model where we need an actual type for the xhtml. Agree that the base model short-circuits this in general. So I take that to mean you have no particular interest. I'll make the change to match R5

**Comment by jamesagnew** [comment]
Yup, as far as I can see, we don't even acknowledge the existence of that class - So no concerns here with any changes to it.
